src/api_routes_http.rs

```diff
@@ -293,93 +278,55 @@ pub fn config(cfg: &mut web::ServiceConfig, rate_limit: &RateLimitCell) {
       )
       // User
       .service(
-        // Account action, I don't like that it's in /user maybe /accounts
-        // Handle /user/register separately to add the register() rate limiter
-        web::resource("/user/register")
-          .guard(guard::Post())
-          .wrap(rate_limit.register())
-          .route(web::post().to(register)),
-      )
-      // User
-      .service(
-        // Handle /user/login separately to add the register() rate limiter
-        // TODO: pretty annoying way to apply rate limits for register and login, we should
-        //       group them under a common path so that rate limit is only applied once (eg under
-        // /account).
-        web::resource("/user/login")
+        web::scope("/account/auth")
           .guard(guard::Post())
           .wrap(rate_limit.register())
-          .route(web::post().to(login)),
-      )
-      .service(
-        web::resource("/user/password_reset")
-          .wrap(rate_limit.register())
-          .route(web::post().to(reset_password)),
-      )
-      .service(
-        // Handle captcha separately
-        web::resource("/user/get_captcha")
-          .wrap(rate_limit.post())
-          .route(web::get().to(get_captcha)),
-      )
-      .service(
-        web::resource("/user/export_settings")
-          .wrap(rate_limit.import_user_settings())
-          .route(web::get().to(export_settings)),
+          .route("register", web::post().to(register))
+          .route("login", web::post().to(login))
+          .route("password_reset", web::post().to(reset_password))
+          .route("get_captcha", web::get().to(get_captcha))
+          .route(
+            "/password_change",
+            web::post().to(change_password_after_reset),
+          )
+          .route("/change_password", web::put().to(change_password))
+          .route("/totp/generate", web::post().to(generate_totp_secret))
+          .route("/totp/update", web::post().to(update_totp)),
       )
       .service(
-        web::resource("/user/import_settings")
+        web::resource("/account/export")
           .wrap(rate_limit.import_user_settings())
+          .route(web::get().to(export_settings))
           .route(web::post().to(import_settings)),
```

Comment:
It needs to have a common path so that the rate limit can apply to both. I changed it to `GET /account/settings/export` and `POST /account/settings/import` now.

Reply:
You could just override the rate limit for those specifically. export / import are actions, not settings, but I don't feel too strongly about that either way.

Reply:
Then it requires two separate blocks like this which is too verbose:
```rust
      .service(
        web::resource("/user/export_settings")
          .wrap(rate_limit.export_user_settings())
          .route(web::get().to(export_settings)),
      )      .service(
        web::resource("/user/import_settings")
          .wrap(rate_limit.import_user_settings())
          .route(web::post().to(import_settings)),
      )
```

Reply:
That's pretty minor imo. They could be put under a `account/takeout` or something tho, especially since we'll eventually need to add a full data export too (even though that won't be importable anywhere).